scheduler/reconcile_util_test.go

```diff
@@ -1329,6 +1331,1327 @@ func TestAllocSet_filterByTainted(t *testing.T) {
 	}
 }
 
+func TestAllocSet_filterByTainted_Disconnected(t *testing.T) {
```

Comment:
It looks like this is a line-by-line recreation of the now-deprecated `TestAllocSet_filterByTainted` above.

We should consider what the maintenance burden of keeping two identical tests like this is. Even once we've marked these fields as deprecated; it'll take some time for them to get fully cycled out (we'll need to keep them in the structs for upgrade safety for several major versions).

I recommend that we create a `testAllocSet_filterByTainted` test function that accepts the 4 jobs under test as parameters, and then call it from two different tests: `TestAllocSet_filterByTainted_Deprecated` and `TestAllocSet_filterByTainted_Disconnected`.

This same comment applies to what we've done in:
* `scheduler/generic_sched_test.go` with `TestServiceSched_Client_Creates_Updates_and_Evals_Disconnect`
* `nomad/node_endpoint_test.go` with `TestClientEndpoint_UpdateStatus_Disconnect_Reconnect`

Reply:
I was thinking about the simplest way to remove the tests in the future, but here is take  2

Reply:
That works! :+1: